perceval/backends/askbot.py

```diff
@@ -0,0 +1,546 @@
+            html_question_items.append(html_question)
+            bs_question = bs4.BeautifulSoup(html_question, "html.parser")
+            tpages = self.ab_parser.parse_number_of_html_pages(bs_question)
+
+            if npages == tpages:
+                next_request = False
+
+            npages = npages + 1
+
+        return html_question_items
+
+    @classmethod
+    def has_resuming(cls):
+        """Returns whether it supports to resume the fetch process.
+
+        :returns: this backend supports items resuming
+        """
+        return True
+
+    @classmethod
+    def has_caching(cls):
+        """Returns whether it supports caching items on the fetch process.
+
+        :returns: this backend does not support items cache
+        """
+        return False
+
+    @staticmethod
+    def metadata_category(item):
+        """Extracts the category from an Askbot item.
+
+        This backend only generates one type of item which is
+        'question'.
+        """
+        return 'question'
+
+    @staticmethod
+    def metadata_id(item):
+        """Extracts the identifier from an Askbot question item."""
+
+        return str(item['id'])
+
+    @staticmethod
+    def metadata_updated_on(item):
+        """Extracts the update time from an Askbot item.
+
+        The timestamp is extracted from 'last_activity_at' field.
+        This date is a UNIX timestamp but needs to be converted to
+        a float value.
+
+        :param item: item generated by the backend
+
+        :returns: a UNIX timestamp
+        """
+        return float(item['last_activity_at'])
+
+    @staticmethod
+    def __build_question(html_question):
+        """Build an Askbot HTML response.
+
+        The method puts together all the information regarding a question
+
+        :param html_question: array of HTML raw pages
+
+        :returns: a dict item with the parsed question information
+        """
+        question_object = {}
+        # Initial case
+        bs_question = bs4.BeautifulSoup(html_question[0], "html.parser")
+        # Parse the user info from the soup container
+        question_container = AskbotParser.parse_question_container(bs_question)
+        # Add the info to the question object
+        question_object.update(question_container)
+        # Add the comments of the question (if any)
+        if AskbotParser.parse_question_comments(bs_question):
+            question_object['comments'] = AskbotParser.parse_question_comments(bs_question)
+
+        answers = AskbotParser.parse_answers(bs_question)
+
+        for page in range(2, len(html_question)+1):
+            position = page - 1
+            try:
+                html_question[position]
+            except IndexError:
+                continue
+            else:
+                bs_question = bs4.BeautifulSoup(html_question[position], "html.parser")
+                answers.extend(AskbotParser.parse_answers(bs_question))
```

Comment:
This call to BeautifulSoup should be part of the parser, not the backend. The backend doesn't need to know about how the parsing is done.